fix(deploy): CDK diff / StaticSiteStack deploy steps omit GITHUB_DEPLOY_ROLE_ARN, so deploy-role IAM grants never reach AWS

## What

Deploy run [73422064458](https://github.com/leonarduk/allotmint/actions/runs/73422064458) failed at the **"Reconcile UiAuthClient OAuth configuration"** step (#3818, just merged) with:

```
An error occurred (AccessDenied) when calling the DescribeUserPoolClient operation: User: arn:aws:sts::152495283689:assumed-role/allotmint-github-deploy/GitHubActions is not authorized to perform: cognito-idp:DescribeUserPoolClient on resource: arn:aws:cognito-idp:***:152495283689:userpool/***_6twGu79Pz because no identity-based policy allows the cognito-idp:DescribeUserPoolClient action
##[error]Process completed with exit code 255.
```

## Root cause

`cdk/stacks/static_site_stack.py` (lines ~416-467) only adds the deploy-role IAM grants — `iam:SimulatePrincipalPolicy`, `cloudformation:CreateChangeSet`/`DescribeChangeSet`/`DeleteChangeSet`, and `cognito-idp:DescribeUserPoolClient`/`UpdateUserPoolClient` — when `os.getenv("GITHUB_DEPLOY_ROLE_ARN")` is non-empty at synth time:

```python
github_deploy_role_arn = os.getenv("GITHUB_DEPLOY_ROLE_ARN", "")
if github_deploy_role_arn:
    ...
```

But in `.github/workflows/deploy-lambda.yml`, **`GITHUB_DEPLOY_ROLE_ARN` is only set in the env block of the "Deploy BackendLambdaStack" step (line 343)**. The two steps that synthesize/deploy `StaticSiteStack` — "CDK diff" (lines 289-295) and "Deploy StaticSiteStack auth resources" (lines 298-304) — do **not** set it. So for those steps, `github_deploy_role_arn` is always `""`, the whole `if` block is skipped, and these grants are never present in the synthesized `StaticSiteStack` template that actually gets deployed.

This explains the symptoms in run 73422064458 and in #3791:
- `cdk diff` and `cdk deploy StaticSiteStack` both synthesize the *same* (grant-less) template every run, so CloudFormation always reports `StaticSiteStack (no changes)` — even though the source code has had these grants since #3210/#3227 (for `iam:SimulatePrincipalPolicy`) and #3802/#3818 (for the cognito-idp grants).
- Step 13 ("Pre-flight IAM permission check") permanently warns that `iam:SimulatePrincipalPolicy` is unavailable and skips itself (degrades gracefully, by design).
- Step 24 ("Reconcile UiAuthClient OAuth configuration", new in #3818) is **not** designed to degrade gracefully — it hard-fails the whole deploy job with exit 255.

`cdk/tests/test_static_site_stack.py::test_github_deploy_role_gets_simulate_principal_policy` and the cognito-idp grant tests pass because the unit tests set `GITHUB_DEPLOY_ROLE_ARN` directly for synthesis — they don't catch that the CI workflow never sets it for the StaticSiteStack-related steps.

## Why this is the critical path

This is a regression introduced (made fatal) by #3818: the missing grant existed silently for weeks (tracked as #3791, masquerading as an account/SCP-level mystery), but #3818 added a step that *requires* one of these never-deployed grants and hard-fails the deploy if it's missing. **Every deploy now fails at step 24.**

## How

In `.github/workflows/deploy-lambda.yml`, add `GITHUB_DEPLOY_ROLE_ARN: ${{ secrets.AWS_ROLE_TO_ASSUME }}` (matching line 343) to the `env:` blocks of:
- "CDK diff" (~line 289-294)
- "Deploy StaticSiteStack auth resources" (~line 298-303)

so the synthesized `StaticSiteStack` template includes the grants, and `cdk deploy StaticSiteStack` actually applies them (no longer "no changes").

## Constraints

- Verify `cdk diff` output for `StaticSiteStack` after this change shows the new IAM statements being added (not "no changes" / not the "mangled non-ASCII" omission seen in run 73422064458).
- Don't widen step 13's failure mode — once the grant actually deploys, the existing pre-flight check should start passing on its own.

## Success

- `cdk deploy StaticSiteStack` in CI shows the IAM Statement Changes for `allotmint-github-deploy` (SimulatePrincipalPolicy, changeset perms, cognito-idp Describe/UpdateUserPoolClient) being added.
- Step 24 ("Reconcile UiAuthClient OAuth configuration") succeeds.
- Step 13 ("Pre-flight IAM permission check") no longer warns about `iam:SimulatePrincipalPolicy` being unavailable.
- #3791 can be closed (its premise — that the CDK template already includes these grants in the deployed StaticSiteStack — was incorrect).

## Failure

If, after this fix, `cdk deploy StaticSiteStack` still shows "no changes" for the IAM grants, re-open the AWS-account-level investigation in #3791 (permission boundary / SCP).

## LLM tier

sonnet

## Related

- #3791 (likely resolved/superseded by this fix — root cause was a workflow env var, not an account-level SCP/boundary issue)
- #3818 (introduced the hard-failing step that surfaced this long-standing gap)
- #3208 / #3210 / #3227 (original SimulatePrincipalPolicy grant, never actually deployed for StaticSiteStack steps)

--- a/.github/workflows/deploy-lambda.yml
+++ b/.github/workflows/deploy-lambda.yml
@@ -263,15 +263,23 @@ jobs:
           DATA_BUCKET: ${{ secrets.DATA_BUCKET }}
           JWT_SECRET: ${{ secrets.JWT_SECRET }}
           GOOGLE_CLIENT_ID: ${{ secrets.GOOGLE_CLIENT_ID }}
+          GITHUB_DEPLOY_ROLE_ARN: ${{ secrets.AWS_ROLE_TO_ASSUME }}
           CDK_OUTDIR: /tmp/cdk.out.${{ github.run_id }}.${{ github.run_attempt }}
         run: npx cdk diff BackendLambdaStack StaticSiteStack --method=template -c prod=true
         working-directory: cdk
 
       - name: Deploy StaticSiteStack auth resources
+        # GITHUB_DEPLOY_ROLE_ARN must be set here so CDK can synthesize the
+        # deploy-role IAM grants in static_site_stack.py (SimulatePrincipalPolicy,
+        # changeset perms, cognito-idp Describe/UpdateUserPoolClient). Without it,
+        # those grants are silently omitted from the deployed StaticSiteStack
+        # template and every dependent step (e.g. "Reconcile UiAuthClient OAuth
+        # configuration") fails with AccessDenied. See #3846.
         env:
           DATA_BUCKET: ${{ secrets.DATA_BUCKET }}
           JWT_SECRET: ${{ secrets.JWT_SECRET }}
           GOOGLE_CLIENT_ID: ${{ secrets.GOOGLE_CLIENT_ID }}
+          GITHUB_DEPLOY_ROLE_ARN: ${{ secrets.AWS_ROLE_TO_ASSUME }}
           CDK_OUTDIR: /tmp/cdk.out.${{ github.run_id }}.${{ github.run_attempt }}
         run: npx cdk deploy StaticSiteStack --require-approval never -c prod=true
         working-directory: cdk

